### PR TITLE
chore: add dapp favouriting remote config and redux variable

### PR DESCRIPTION
**Pull request by kathaypacific**
### Description

As the title

### Other changes

N/A

### Tested

N/A

### How others should test

N/A

### Related issues

- Fixes RET-522

### Backwards compatibility

N/A

### CODE REVIEW

**Review by jeanregisser — APPROVED** [review]
👍

**Review by jacob-israel-turner — APPROVED** [review]
LGTM!

One thing to note - if we remove a dapp from the dapplist api, but a user has previously favorited it, then it will still appear in the `favouriteDapps` list. I only imagine this will be an issue if we want to remove a dapp for abuse, which is hopefully rare.

When actually rendering the favorited dapps, we might just want to check that each item still exists in the base list. Alternatively, we could make `favoriteDapps` a list of IDs, and require the list component to fetch the actual dapp data from `dappsList`.

**Comment by kathaypacific** [comment]
> LGTM!
> 
> One thing to note - if we remove a dapp from the dapplist api, but a user has previously favorited it, then it will still appear in the `favouriteDapps` list. I only imagine this will be an issue if we want to remove a dapp for abuse, which is hopefully rare.
> 
> When actually rendering the favorited dapps, we might just want to check that each item still exists in the base list. Alternatively, we could make `favoriteDapps` a list of IDs, and require the list component to fetch the actual dapp data from `dappsList`.

Yeah good shout, it's on my radar to add something to the `fetchDappsListCompleted` reducer to remove any recently used / favourited dapps after the list is fetched :)

**Comment by kathaypacific** [comment]
> LGTM!
> 
> One thing to note - if we remove a dapp from the dapplist api, but a user has previously favorited it, then it will still appear in the `favouriteDapps` list. I only imagine this will be an issue if we want to remove a dapp for abuse, which is hopefully rare.
> 
> When actually rendering the favorited dapps, we might just want to check that each item still exists in the base list. Alternatively, we could make `favoriteDapps` a list of IDs, and require the list component to fetch the actual dapp data from `dappsList`.

I decided to bite the bullet and address the comment here https://github.com/valora-inc/wallet/pull/3248 - the context for why the recently used dapps didn't use an id in the first place, is that we weren't storing the dapps list at all in redux before. i had left a TODO comment to make this exact change but forgot about it, thanks for raising it again

File: src/app/saga.ts [hunk 1]
```diff
@@ -210,6 +210,7 @@ export interface RemoteConfigValues {
   showGuidedOnboardingCopy: boolean
   centralPhoneVerificationEnabled: boolean
   networkTimeoutSeconds: number
+  dappFavouritesEnabled: boolean
```
[the remark this answers is not in the document]
Comment (kathaypacific) [reply src/app/saga.ts:213]: :( i want to refactor the whole project to use british english, it just looks more right!

**Pull request merged**